hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java

```diff
@@ -201,23 +209,31 @@ private void failInit(Channel ch, IOException e) {
     // fail all pending calls
     ch.pipeline().fireUserEventTriggered(BufferCallEvent.fail(e));
     shutdown0();
+    rpcClient.failedServers.addToFailedServers(remoteId.getAddress(), e);
```

Comment:
We have this line right after the failInit call when connecting fail, it is at line 334 in the old code. I moved it here since we will call failInit at other places, I think for all those cases we should add the server to failedServers.